app/db/AtomWorkshopDB.scala

```diff
@@ -1,19 +1,22 @@
 package db
 
 import cats.syntax.either._
-import com.gu.atom.data.{DataStoreResultUtil, DynamoDataStore, IDNotFound}
+import com.gu.atom.data.{DataStoreResultUtil, DynamoDataStore, IDNotFound, VersionConflictError}
 import com.gu.contentatom.thrift.{Atom, AtomType}
 import com.gu.pandomainauth.model.User
-import models.{AtomAPIError, AtomWorkshopDynamoDatastoreError, UnknownAtomError}
+import models.{AtomAPIError, AtomWorkshopDynamoConflictError, AtomWorkshopDynamoDatastoreError, UnknownAtomError}
 import play.api.Logging
 import util.AtomLogic._
 
-trait AtomWorkshopDBAPI {
+trait AtomWorkshopDBAPI extends Logging {
 
   def transformAtomLibResult[T](atomType: AtomType, id: String, result: DataStoreResultUtil.DataStoreResult[T]): Either[AtomAPIError, T] = result match {
     case Left(IDNotFound) => Left(UnknownAtomError(atomType, id))
+    case Left(VersionConflictError(_)) => Left(AtomWorkshopDynamoConflictError)
     case Left(e) => Left(AtomWorkshopDynamoDatastoreError(e.msg))
-    case Right(r) => Right(r)
+    case Right(r) =>
+      logger.info(s"Successfully updated atom of type ${atomType.name} with id $id")
```

Comment:
This logging has been moved so it only gets logged if the atom was actually updated